cli/js/lib.deno.ns.d.ts

```diff
@@ -35,69 +35,67 @@ declare namespace Deno {
    * when `Deno.runTests` is used */
   export function test(name: string, fn: TestFunction): void;
 
-  enum TestStatus {
-    Passed = "passed",
-    Failed = "failed",
-    Ignored = "ignored"
-  }
-
-  interface TestResult {
-    name: string;
-    status: TestStatus;
-    duration?: number;
-    error?: Error;
-  }
-
-  interface TestStats {
-    filtered: number;
-    ignored: number;
-    measured: number;
-    passed: number;
-    failed: number;
-  }
-
-  export enum TestEvent {
+  enum TestEvent {
     Start = "start",
     TestStart = "testStart",
     TestEnd = "testEnd",
     End = "end"
   }
 
-  interface TestEventStart {
-    kind: TestEvent.Start;
-    tests: number;
-  }
-
-  interface TestEventTestStart {
-    kind: TestEvent.TestStart;
-    name: string;
-  }
-
-  interface TestEventTestEnd {
-    kind: TestEvent.TestEnd;
-    result: TestResult;
-  }
-
-  interface TestEventEnd {
-    kind: TestEvent.End;
-    stats: TestStats;
-    duration: number;
-    results: TestResult[];
-  }
-
-  interface TestReporter {
-    start(event: TestEventStart): Promise<void>;
-    testStart(msg: TestEventTestStart): Promise<void>;
-    testEnd(msg: TestEventTestEnd): Promise<void>;
-    end(event: TestEventEnd): Promise<void>;
-  }
-
-  export class ConsoleTestReporter implements TestReporter {
-    constructor();
-    start(event: TestEventStart): Promise<void>;
-    testStart(msg: TestEventTestStart): Promise<void>;
-    testEnd(msg: TestEventTestEnd): Promise<void>;
-    end(event: TestEventEnd): Promise<void>;
+  export namespace tests {
+    export enum Status {
+      Passed = "passed",
+      Failed = "failed",
+      Ignored = "ignored"
+    }
+
+    export interface Result {
+      name: string;
+      status: Status;
+      duration: number;
+      error?: Error;
+    }
+
+    export interface StartMessage {
+      tests: TestDefinition[];
+    }
+
+    export interface TestStartMessage {
+      test: TestDefinition;
+    }
+
+    export interface TestEndMessage {
+      result: Result;
+    }
+
+    export interface EndMessage {
+      stats: Stats;
+      duration: number;
+      results: Result[];
+    }
+
+    export interface Stats {
+      filtered: number;
+      ignored: number;
+      measured: number;
+      passed: number;
+      failed: number;
+    }
+
+    export interface Reporter {
+      start(message: StartMessage): Promise<void>;
+      testStart(message: TestStartMessage): Promise<void>;
+      testEnd(message: TestEndMessage): Promise<void>;
+      end(message: EndMessage): Promise<void>;
+    }
+
+    export class ConsoleReporter implements Reporter {
+      constructor();
+      start(message: StartMessage): Promise<void>;
+      testStart(message: TestStartMessage): Promise<void>;
+      testEnd(message: TestEndMessage): Promise<void>;
+      end(message: EndMessage): Promise<void>;
+    }
```

Comment:
I don't like that the test interface has become so complex that we now need to introduce a namespace to keep all the associated classes and interfaces.

It used to be just a single function.

Maybe we should work on simplifying it rather than resorting to namespaces.

Also should it be "tests" or "testing"? In std we have "std/testing".

Reply:
The test interface could be simplified greatly. Reporter could be a single function instead of a class. All of the message types could be reduced into one.

Reply:
ConsoleReporter doesn't need to be exposed to the public at all.